PYIC-5874: Headless CRI stub

Wynndow opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Allow the CRI stub to be invoked via an API, rather than having to use the web form.

### Why did it change

To allow us to test core-back journeys just using API calls - this will allow us to separate testing back end and front end logic.

### Sample inputs

These inputs have been run against the DCMAW stub (configured to act like the F2F CRI to allow VCs to be sent to a queue), running locally in a container, using core's local-running setup.

```
// Happy request
curl http://localhost:3003/api/authorize -v -d '{
  "clientId": "ipv-core-dev01",
  "request": "eyJjdHkiOiJKV1QiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiUlNBLU9BRVAtMjU2In0.G-7NVw-U-y_nuqhI_atQchaR62I2428w-jquIZnQkYkjmJJgS0V6-23TBwN3jNftrE3byisNFnje9zXf7hBBgr7UfnizoiSZyVufocDvt33CrHJdHuUEracOfXqhxwBFmmK0Qbwp6-FEKuxMQwUpDCH64r0M316jw7OycKIVmThA0amzhDhPWlVFqMT68GauwW6ajR8y9OYngzsUmZrBUBabZDdrZeqo8Hr95P0AwHoCD6i4EoTo6DPn9nl3L7Jyw1TaWguk0JnRPiU0MxLA8ks7ryXqXVBdx1breGxzUS4RMlbEX7mn5hPUDiD6wGlWbi2jPW8_5xMV-ZhoCQlufg.Ybu9nJra4aje5OEG.x4wrisYRGg43isZqbYRVP3EY8NpYmmffE_QJVbwuK3dsHPDqp5wsqcen3Dbg5GW--7OjzDpOnkoX3-oWJdnrL5n2Jk42OtXOGH_tUrR4qV2Vc9EdKE8WYvQieH1-VF5XoUQG-ltBk5B-tLZ18lC8w9Zy1PR_cfetYH35CcK7wAUaoX5cOw_aMoao2go2oW5DDD-9jj1yj0DjuOovswum-YFyZYmKizOxtLRiw1RsKR2KtQ5ZrWiac4KIAftLLJGn4gBVAkj6D9JbTacIl1hxQulqN4kSsZ0EKpTpnmNjJ1x8E6yqAT2uOEDjskg3gzD4wOwvrDh1sjjQywOOxbzE6W4uL8vcDn2pqhja-_M_FKsWWusSkCSMM5HYuo7-fOoRKRnLedKiD5uP3WWRGj91Qez_QMLvQNgWQ6lvqLNHmuAWaj7rFQCYwG0LkeNrP28qgUsmaGvxLkakJDvjAMBxnWXi-He5Nk-bMifUEikZl3hM5rDseBv5BbuZ0N_7DOAaUJ3LS1JNBIbN_11wr05kKRi0JG3j1yNsOd15Si-N24DYWrCgzrY4My2Aibv-jzbO_tHSg307p9PQAj9lSVwbMvb6PHFRz_MbHOCYabvuLvl47Le6lCy4JIWXHIb8jHReJmABCSAf231--00x6n8UayZfKBD_T-M6JX1gR3OhoNNtvu4q34q9pKTzPS-uKtE7uj-zDXb1JA8IfyMAkdXzkTjYNmlJrE4YmE-ho07_fiC8XlYE9mofx2gHC5Bjze-agH2jIaofbEtuVXGU6-M84UyIvCCJWzTO7-jKjmZHpFAbUm18Lj_Ow1sMJZlwvvP2HdrpkeLYx6d92g-o6xsuNhc0gxzatJDE60Gx8Or4QRfR5Ras-fsM6qIyF3fu1JRwJi2UNqwrNn9AJNSKa0n7N1dBV6bou91iTOkUbIacC4X8UBATF-mLlVv7NqVaw3rrUhtD0eS1xRjVn2P4Kg5yf6zEMquTJbYKL-uEZjNALg32u_Rr9APRayoBeIAKhlPlpwo0YEKariBYDooDNI_hh2-HlAn0ZVKJ8ugqBkaYIc4VMmqBmjQ.nQtGZi9sBcYxewmKlDTvtg",
  "credentialSubjectJson": "{\"passport\":[{\"expiryDate\":\"2030-01-01\",\"icaoIssuerCode\":\"GBR\",\"documentNumber\":\"321654987\"}],\"name\":[{\"nameParts\":[{\"type\":\"GivenName\",\"value\":\"Kenneth\"},{\"type\":\"FamilyName\",\"value\":\"Decerqueira\"}]}],\"birthDate\":[{\"value\":\"1965-07-08\"}]}",
  "evidenceJson": "{\"activityHistoryScore\":1,\"checkDetails\":[{\"checkMethod\":\"vri\"},{\"biometricVerificationProcessLevel\":3,\"checkMethod\":\"bvr\"}],\"validityScore\":2,\"strengthScore\":3,\"type\":\"IdentityCheck\"}",
  "resourceId": "something"
}'

// Request with CI in evidence blob
curl http://localhost:3003/api/authorize -v -d '{
"clientId": "ipv-core-dev01",
"request": "eyJjdHkiOiJKV1QiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiUlNBLU9BRVAtMjU2In0.G-7NVw-U-y_nuqhI_atQchaR62I2428w-jquIZnQkYkjmJJgS0V6-23TBwN3jNftrE3byisNFnje9zXf7hBBgr7UfnizoiSZyVufocDvt33CrHJdHuUEracOfXqhxwBFmmK0Qbwp6-FEKuxMQwUpDCH64r0M316jw7OycKIVmThA0amzhDhPWlVFqMT68GauwW6ajR8y9OYngzsUmZrBUBabZDdrZeqo8Hr95P0AwHoCD6i4EoTo6DPn9nl3L7Jyw1TaWguk0JnRPiU0MxLA8ks7ryXqXVBdx1breGxzUS4RMlbEX7mn5hPUDiD6wGlWbi2jPW8_5xMV-ZhoCQlufg.Ybu9nJra4aje5OEG.x4wrisYRGg43isZqbYRVP3EY8NpYmmffE_QJVbwuK3dsHPDqp5wsqcen3Dbg5GW--7OjzDpOnkoX3-oWJdnrL5n2Jk42OtXOGH_tUrR4qV2Vc9EdKE8WYvQieH1-VF5XoUQG-ltBk5B-tLZ18lC8w9Zy1PR_cfetYH35CcK7wAUaoX5cOw_aMoao2go2oW5DDD-9jj1yj0DjuOovswum-YFyZYmKizOxtLRiw1RsKR2KtQ5ZrWiac4KIAftLLJGn4gBVAkj6D9JbTacIl1hxQulqN4kSsZ0EKpTpnmNjJ1x8E6yqAT2uOEDjskg3gzD4wOwvrDh1sjjQywOOxbzE6W4uL8vcDn2pqhja-_M_FKsWWusSkCSMM5HYuo7-fOoRKRnLedKiD5uP3WWRGj91Qez_QMLvQNgWQ6lvqLNHmuAWaj7rFQCYwG0LkeNrP28qgUsmaGvxLkakJDvjAMBxnWXi-He5Nk-bMifUEikZl3hM5rDseBv5BbuZ0N_7DOAaUJ3LS1JNBIbN_11wr05kKRi0JG3j1yNsOd15Si-N24DYWrCgzrY4My2Aibv-jzbO_tHSg307p9PQAj9lSVwbMvb6PHFRz_MbHOCYabvuLvl47Le6lCy4JIWXHIb8jHReJmABCSAf231--00x6n8UayZfKBD_T-M6JX1gR3OhoNNtvu4q34q9pKTzPS-uKtE7uj-zDXb1JA8IfyMAkdXzkTjYNmlJrE4YmE-ho07_fiC8XlYE9mofx2gHC5Bjze-agH2jIaofbEtuVXGU6-M84UyIvCCJWzTO7-jKjmZHpFAbUm18Lj_Ow1sMJZlwvvP2HdrpkeLYx6d92g-o6xsuNhc0gxzatJDE60Gx8Or4QRfR5Ras-fsM6qIyF3fu1JRwJi2UNqwrNn9AJNSKa0n7N1dBV6bou91iTOkUbIacC4X8UBATF-mLlVv7NqVaw3rrUhtD0eS1xRjVn2P4Kg5yf6zEMquTJbYKL-uEZjNALg32u_Rr9APRayoBeIAKhlPlpwo0YEKariBYDooDNI_hh2-HlAn0ZVKJ8ugqBkaYIc4VMmqBmjQ.nQtGZi9sBcYxewmKlDTvtg",
"credentialSubjectJson": "{\"passport\":[{\"expiryDate\":\"2030-01-01\",\"icaoIssuerCode\":\"GBR\",\"documentNumber\":\"321654987\"}],\"name\":[{\"nameParts\":[{\"type\":\"GivenName\",\"value\":\"Kenneth\"},{\"type\":\"FamilyName\",\"value\":\"Decerqueira\"}]}],\"birthDate\":[{\"value\":\"1965-07-08\"}]}",
"evidenceJson": "{\"failedCheckDetails\":[{\"identityCheckPolicy\":\"published\",\"checkMethod\":\"vcrypt\"},{\"biometricVerificationProcessLevel\":3,\"checkMethod\":\"bvr\"}],\"validityScore\":0,\"strengthScore\":4,\"ci\":[\"D06\",\"D05\"],\"type\":\"IdentityCheck\"}",
"resourceId": "something"
}'

// Request with mitigations
curl http://localhost:3003/api/authorize -v -d '{
  "clientId": "ipv-core-dev01",
  "request": "eyJjdHkiOiJKV1QiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiUlNBLU9BRVAtMjU2In0.G-7NVw-U-y_nuqhI_atQchaR62I2428w-jquIZnQkYkjmJJgS0V6-23TBwN3jNftrE3byisNFnje9zXf7hBBgr7UfnizoiSZyVufocDvt33CrHJdHuUEracOfXqhxwBFmmK0Qbwp6-FEKuxMQwUpDCH64r0M316jw7OycKIVmThA0amzhDhPWlVFqMT68GauwW6ajR8y9OYngzsUmZrBUBabZDdrZeqo8Hr95P0AwHoCD6i4EoTo6DPn9nl3L7Jyw1TaWguk0JnRPiU0MxLA8ks7ryXqXVBdx1breGxzUS4RMlbEX7mn5hPUDiD6wGlWbi2jPW8_5xMV-ZhoCQlufg.Ybu9nJra4aje5OEG.x4wrisYRGg43isZqbYRVP3EY8NpYmmffE_QJVbwuK3dsHPDqp5wsqcen3Dbg5GW--7OjzDpOnkoX3-oWJdnrL5n2Jk42OtXOGH_tUrR4qV2Vc9EdKE8WYvQieH1-VF5XoUQG-ltBk5B-tLZ18lC8w9Zy1PR_cfetYH35CcK7wAUaoX5cOw_aMoao2go2oW5DDD-9jj1yj0DjuOovswum-YFyZYmKizOxtLRiw1RsKR2KtQ5ZrWiac4KIAftLLJGn4gBVAkj6D9JbTacIl1hxQulqN4kSsZ0EKpTpnmNjJ1x8E6yqAT2uOEDjskg3gzD4wOwvrDh1sjjQywOOxbzE6W4uL8vcDn2pqhja-_M_FKsWWusSkCSMM5HYuo7-fOoRKRnLedKiD5uP3WWRGj91Qez_QMLvQNgWQ6lvqLNHmuAWaj7rFQCYwG0LkeNrP28qgUsmaGvxLkakJDvjAMBxnWXi-He5Nk-bMifUEikZl3hM5rDseBv5BbuZ0N_7DOAaUJ3LS1JNBIbN_11wr05kKRi0JG3j1yNsOd15Si-N24DYWrCgzrY4My2Aibv-jzbO_tHSg307p9PQAj9lSVwbMvb6PHFRz_MbHOCYabvuLvl47Le6lCy4JIWXHIb8jHReJmABCSAf231--00x6n8UayZfKBD_T-M6JX1gR3OhoNNtvu4q34q9pKTzPS-uKtE7uj-zDXb1JA8IfyMAkdXzkTjYNmlJrE4YmE-ho07_fiC8XlYE9mofx2gHC5Bjze-agH2jIaofbEtuVXGU6-M84UyIvCCJWzTO7-jKjmZHpFAbUm18Lj_Ow1sMJZlwvvP2HdrpkeLYx6d92g-o6xsuNhc0gxzatJDE60Gx8Or4QRfR5Ras-fsM6qIyF3fu1JRwJi2UNqwrNn9AJNSKa0n7N1dBV6bou91iTOkUbIacC4X8UBATF-mLlVv7NqVaw3rrUhtD0eS1xRjVn2P4Kg5yf6zEMquTJbYKL-uEZjNALg32u_Rr9APRayoBeIAKhlPlpwo0YEKariBYDooDNI_hh2-HlAn0ZVKJ8ugqBkaYIc4VMmqBmjQ.nQtGZi9sBcYxewmKlDTvtg",
  "credentialSubjectJson": "{\"passport\":[{\"expiryDate\":\"2030-01-01\",\"icaoIssuerCode\":\"GBR\",\"documentNumber\":\"321654987\"}],\"name\":[{\"nameParts\":[{\"type\":\"GivenName\",\"value\":\"Kenneth\"},{\"type\":\"FamilyName\",\"value\":\"Decerqueira\"}]}],\"birthDate\":[{\"value\":\"1965-07-08\"}]}",
  "evidenceJson": "{\"activityHistoryScore\":1,\"checkDetails\":[{\"checkMethod\":\"vri\"},{\"biometricVerificationProcessLevel\":3,\"checkMethod\":\"bvr\"}],\"validityScore\":2,\"strengthScore\":3,\"type\":\"IdentityCheck\"}",
  "resourceId": "something",
  "mitigations": {
    "mitigatedCi": ["D01"],
    "cimitStubUrl": "https://cimit.stubs.account.gov.uk",
    "cimitStubApiKey": "anAPIKey"
  }
}'

// Request with NBF
curl http://localhost:3003/api/authorize -v -d '{
  "clientId": "ipv-core-dev01",
  "request": "eyJjdHkiOiJKV1QiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiUlNBLU9BRVAtMjU2In0.G-7NVw-U-y_nuqhI_atQchaR62I2428w-jquIZnQkYkjmJJgS0V6-23TBwN3jNftrE3byisNFnje9zXf7hBBgr7UfnizoiSZyVufocDvt33CrHJdHuUEracOfXqhxwBFmmK0Qbwp6-FEKuxMQwUpDCH64r0M316jw7OycKIVmThA0amzhDhPWlVFqMT68GauwW6ajR8y9OYngzsUmZrBUBabZDdrZeqo8Hr95P0AwHoCD6i4EoTo6DPn9nl3L7Jyw1TaWguk0JnRPiU0MxLA8ks7ryXqXVBdx1breGxzUS4RMlbEX7mn5hPUDiD6wGlWbi2jPW8_5xMV-ZhoCQlufg.Ybu9nJra4aje5OEG.x4wrisYRGg43isZqbYRVP3EY8NpYmmffE_QJVbwuK3dsHPDqp5wsqcen3Dbg5GW--7OjzDpOnkoX3-oWJdnrL5n2Jk42OtXOGH_tUrR4qV2Vc9EdKE8WYvQieH1-VF5XoUQG-ltBk5B-tLZ18lC8w9Zy1PR_cfetYH35CcK7wAUaoX5cOw_aMoao2go2oW5DDD-9jj1yj0DjuOovswum-YFyZYmKizOxtLRiw1RsKR2KtQ5ZrWiac4KIAftLLJGn4gBVAkj6D9JbTacIl1hxQulqN4kSsZ0EKpTpnmNjJ1x8E6yqAT2uOEDjskg3gzD4wOwvrDh1sjjQywOOxbzE6W4uL8vcDn2pqhja-_M_FKsWWusSkCSMM5HYuo7-fOoRKRnLedKiD5uP3WWRGj91Qez_QMLvQNgWQ6lvqLNHmuAWaj7rFQCYwG0LkeNrP28qgUsmaGvxLkakJDvjAMBxnWXi-He5Nk-bMifUEikZl3hM5rDseBv5BbuZ0N_7DOAaUJ3LS1JNBIbN_11wr05kKRi0JG3j1yNsOd15Si-N24DYWrCgzrY4My2Aibv-jzbO_tHSg307p9PQAj9lSVwbMvb6PHFRz_MbHOCYabvuLvl47Le6lCy4JIWXHIb8jHReJmABCSAf231--00x6n8UayZfKBD_T-M6JX1gR3OhoNNtvu4q34q9pKTzPS-uKtE7uj-zDXb1JA8IfyMAkdXzkTjYNmlJrE4YmE-ho07_fiC8XlYE9mofx2gHC5Bjze-agH2jIaofbEtuVXGU6-M84UyIvCCJWzTO7-jKjmZHpFAbUm18Lj_Ow1sMJZlwvvP2HdrpkeLYx6d92g-o6xsuNhc0gxzatJDE60Gx8Or4QRfR5Ras-fsM6qIyF3fu1JRwJi2UNqwrNn9AJNSKa0n7N1dBV6bou91iTOkUbIacC4X8UBATF-mLlVv7NqVaw3rrUhtD0eS1xRjVn2P4Kg5yf6zEMquTJbYKL-uEZjNALg32u_Rr9APRayoBeIAKhlPlpwo0YEKariBYDooDNI_hh2-HlAn0ZVKJ8ugqBkaYIc4VMmqBmjQ.nQtGZi9sBcYxewmKlDTvtg",
  "credentialSubjectJson": "{\"passport\":[{\"expiryDate\":\"2030-01-01\",\"icaoIssuerCode\":\"GBR\",\"documentNumber\":\"321654987\"}],\"name\":[{\"nameParts\":[{\"type\":\"GivenName\",\"value\":\"Kenneth\"},{\"type\":\"FamilyName\",\"value\":\"Decerqueira\"}]}],\"birthDate\":[{\"value\":\"1965-07-08\"}]}",
  "evidenceJson": "{\"activityHistoryScore\":1,\"checkDetails\":[{\"checkMethod\":\"vri\"},{\"biometricVerificationProcessLevel\":3,\"checkMethod\":\"bvr\"}],\"validityScore\":2,\"strengthScore\":3,\"type\":\"IdentityCheck\"}",
  "resourceId": "something",
  "nbf": 1714577018
}'

// Request to send VC to async queue
curl http://localhost:3003/api/authorize -v -d '{
  "clientId": "ipv-core-dev01",
  "request": "eyJjdHkiOiJKV1QiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiUlNBLU9BRVAtMjU2In0.G-7NVw-U-y_nuqhI_atQchaR62I2428w-jquIZnQkYkjmJJgS0V6-23TBwN3jNftrE3byisNFnje9zXf7hBBgr7UfnizoiSZyVufocDvt33CrHJdHuUEracOfXqhxwBFmmK0Qbwp6-FEKuxMQwUpDCH64r0M316jw7OycKIVmThA0amzhDhPWlVFqMT68GauwW6ajR8y9OYngzsUmZrBUBabZDdrZeqo8Hr95P0AwHoCD6i4EoTo6DPn9nl3L7Jyw1TaWguk0JnRPiU0MxLA8ks7ryXqXVBdx1breGxzUS4RMlbEX7mn5hPUDiD6wGlWbi2jPW8_5xMV-ZhoCQlufg.Ybu9nJra4aje5OEG.x4wrisYRGg43isZqbYRVP3EY8NpYmmffE_QJVbwuK3dsHPDqp5wsqcen3Dbg5GW--7OjzDpOnkoX3-oWJdnrL5n2Jk42OtXOGH_tUrR4qV2Vc9EdKE8WYvQieH1-VF5XoUQG-ltBk5B-tLZ18lC8w9Zy1PR_cfetYH35CcK7wAUaoX5cOw_aMoao2go2oW5DDD-9jj1yj0DjuOovswum-YFyZYmKizOxtLRiw1RsKR2KtQ5ZrWiac4KIAftLLJGn4gBVAkj6D9JbTacIl1hxQulqN4kSsZ0EKpTpnmNjJ1x8E6yqAT2uOEDjskg3gzD4wOwvrDh1sjjQywOOxbzE6W4uL8vcDn2pqhja-_M_FKsWWusSkCSMM5HYuo7-fOoRKRnLedKiD5uP3WWRGj91Qez_QMLvQNgWQ6lvqLNHmuAWaj7rFQCYwG0LkeNrP28qgUsmaGvxLkakJDvjAMBxnWXi-He5Nk-bMifUEikZl3hM5rDseBv5BbuZ0N_7DOAaUJ3LS1JNBIbN_11wr05kKRi0JG3j1yNsOd15Si-N24DYWrCgzrY4My2Aibv-jzbO_tHSg307p9PQAj9lSVwbMvb6PHFRz_MbHOCYabvuLvl47Le6lCy4JIWXHIb8jHReJmABCSAf231--00x6n8UayZfKBD_T-M6JX1gR3OhoNNtvu4q34q9pKTzPS-uKtE7uj-zDXb1JA8IfyMAkdXzkTjYNmlJrE4YmE-ho07_fiC8XlYE9mofx2gHC5Bjze-agH2jIaofbEtuVXGU6-M84UyIvCCJWzTO7-jKjmZHpFAbUm18Lj_Ow1sMJZlwvvP2HdrpkeLYx6d92g-o6xsuNhc0gxzatJDE60Gx8Or4QRfR5Ras-fsM6qIyF3fu1JRwJi2UNqwrNn9AJNSKa0n7N1dBV6bou91iTOkUbIacC4X8UBATF-mLlVv7NqVaw3rrUhtD0eS1xRjVn2P4Kg5yf6zEMquTJbYKL-uEZjNALg32u_Rr9APRayoBeIAKhlPlpwo0YEKariBYDooDNI_hh2-HlAn0ZVKJ8ugqBkaYIc4VMmqBmjQ.nQtGZi9sBcYxewmKlDTvtg",
  "credentialSubjectJson": "{\"passport\":[{\"expiryDate\":\"2030-01-01\",\"icaoIssuerCode\":\"GBR\",\"documentNumber\":\"321654987\"}],\"name\":[{\"nameParts\":[{\"type\":\"GivenName\",\"value\":\"Kenneth\"},{\"type\":\"FamilyName\",\"value\":\"Decerqueira\"}]}],\"birthDate\":[{\"value\":\"1965-07-08\"}]}",
  "evidenceJson": "{\"activityHistoryScore\":1,\"checkDetails\":[{\"checkMethod\":\"vri\"},{\"biometricVerificationProcessLevel\":3,\"checkMethod\":\"bvr\"}],\"validityScore\":2,\"strengthScore\":3,\"type\":\"IdentityCheck\"}",
  "resourceId": "something",
  "f2f": {
    "sendVcToQueue": true,
    "queueName": "stubQueue_F2FQueue_build"
  }
}'

// Request to send error to F2F queue
curl http://localhost:3003/api/authorize -v -d '{
  "clientId": "ipv-core-dev01",
  "request": "eyJjdHkiOiJKV1QiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiUlNBLU9BRVAtMjU2In0.G-7NVw-U-y_nuqhI_atQchaR62I2428w-jquIZnQkYkjmJJgS0V6-23TBwN3jNftrE3byisNFnje9zXf7hBBgr7UfnizoiSZyVufocDvt33CrHJdHuUEracOfXqhxwBFmmK0Qbwp6-FEKuxMQwUpDCH64r0M316jw7OycKIVmThA0amzhDhPWlVFqMT68GauwW6ajR8y9OYngzsUmZrBUBabZDdrZeqo8Hr95P0AwHoCD6i4EoTo6DPn9nl3L7Jyw1TaWguk0JnRPiU0MxLA8ks7ryXqXVBdx1breGxzUS4RMlbEX7mn5hPUDiD6wGlWbi2jPW8_5xMV-ZhoCQlufg.Ybu9nJra4aje5OEG.x4wrisYRGg43isZqbYRVP3EY8NpYmmffE_QJVbwuK3dsHPDqp5wsqcen3Dbg5GW--7OjzDpOnkoX3-oWJdnrL5n2Jk42OtXOGH_tUrR4qV2Vc9EdKE8WYvQieH1-VF5XoUQG-ltBk5B-tLZ18lC8w9Zy1PR_cfetYH35CcK7wAUaoX5cOw_aMoao2go2oW5DDD-9jj1yj0DjuOovswum-YFyZYmKizOxtLRiw1RsKR2KtQ5ZrWiac4KIAftLLJGn4gBVAkj6D9JbTacIl1hxQulqN4kSsZ0EKpTpnmNjJ1x8E6yqAT2uOEDjskg3gzD4wOwvrDh1sjjQywOOxbzE6W4uL8vcDn2pqhja-_M_FKsWWusSkCSMM5HYuo7-fOoRKRnLedKiD5uP3WWRGj91Qez_QMLvQNgWQ6lvqLNHmuAWaj7rFQCYwG0LkeNrP28qgUsmaGvxLkakJDvjAMBxnWXi-He5Nk-bMifUEikZl3hM5rDseBv5BbuZ0N_7DOAaUJ3LS1JNBIbN_11wr05kKRi0JG3j1yNsOd15Si-N24DYWrCgzrY4My2Aibv-jzbO_tHSg307p9PQAj9lSVwbMvb6PHFRz_MbHOCYabvuLvl47Le6lCy4JIWXHIb8jHReJmABCSAf231--00x6n8UayZfKBD_T-M6JX1gR3OhoNNtvu4q34q9pKTzPS-uKtE7uj-zDXb1JA8IfyMAkdXzkTjYNmlJrE4YmE-ho07_fiC8XlYE9mofx2gHC5Bjze-agH2jIaofbEtuVXGU6-M84UyIvCCJWzTO7-jKjmZHpFAbUm18Lj_Ow1sMJZlwvvP2HdrpkeLYx6d92g-o6xsuNhc0gxzatJDE60Gx8Or4QRfR5Ras-fsM6qIyF3fu1JRwJi2UNqwrNn9AJNSKa0n7N1dBV6bou91iTOkUbIacC4X8UBATF-mLlVv7NqVaw3rrUhtD0eS1xRjVn2P4Kg5yf6zEMquTJbYKL-uEZjNALg32u_Rr9APRayoBeIAKhlPlpwo0YEKariBYDooDNI_hh2-HlAn0ZVKJ8ugqBkaYIc4VMmqBmjQ.nQtGZi9sBcYxewmKlDTvtg",
  "credentialSubjectJson": "{\"passport\":[{\"expiryDate\":\"2030-01-01\",\"icaoIssuerCode\":\"GBR\",\"documentNumber\":\"321654987\"}],\"name\":[{\"nameParts\":[{\"type\":\"GivenName\",\"value\":\"Kenneth\"},{\"type\":\"FamilyName\",\"value\":\"Decerqueira\"}]}],\"birthDate\":[{\"value\":\"1965-07-08\"}]}",
  "evidenceJson": "{\"activityHistoryScore\":1,\"checkDetails\":[{\"checkMethod\":\"vri\"},{\"biometricVerificationProcessLevel\":3,\"checkMethod\":\"bvr\"}],\"validityScore\":2,\"strengthScore\":3,\"type\":\"IdentityCheck\"}",
  "resourceId": "something",
  "f2f": {
    "sendErrorToQueue": true,
    "queueName": "stubQueue_F2FQueue_build"
  }
}'

// Request to force an OAuth error from auth endpoint
curl http://localhost:3003/api/authorize -v -d '{
  "clientId": "ipv-core-dev01",
  "request": "eyJjdHkiOiJKV1QiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiUlNBLU9BRVAtMjU2In0.G-7NVw-U-y_nuqhI_atQchaR62I2428w-jquIZnQkYkjmJJgS0V6-23TBwN3jNftrE3byisNFnje9zXf7hBBgr7UfnizoiSZyVufocDvt33CrHJdHuUEracOfXqhxwBFmmK0Qbwp6-FEKuxMQwUpDCH64r0M316jw7OycKIVmThA0amzhDhPWlVFqMT68GauwW6ajR8y9OYngzsUmZrBUBabZDdrZeqo8Hr95P0AwHoCD6i4EoTo6DPn9nl3L7Jyw1TaWguk0JnRPiU0MxLA8ks7ryXqXVBdx1breGxzUS4RMlbEX7mn5hPUDiD6wGlWbi2jPW8_5xMV-ZhoCQlufg.Ybu9nJra4aje5OEG.x4wrisYRGg43isZqbYRVP3EY8NpYmmffE_QJVbwuK3dsHPDqp5wsqcen3Dbg5GW--7OjzDpOnkoX3-oWJdnrL5n2Jk42OtXOGH_tUrR4qV2Vc9EdKE8WYvQieH1-VF5XoUQG-ltBk5B-tLZ18lC8w9Zy1PR_cfetYH35CcK7wAUaoX5cOw_aMoao2go2oW5DDD-9jj1yj0DjuOovswum-YFyZYmKizOxtLRiw1RsKR2KtQ5ZrWiac4KIAftLLJGn4gBVAkj6D9JbTacIl1hxQulqN4kSsZ0EKpTpnmNjJ1x8E6yqAT2uOEDjskg3gzD4wOwvrDh1sjjQywOOxbzE6W4uL8vcDn2pqhja-_M_FKsWWusSkCSMM5HYuo7-fOoRKRnLedKiD5uP3WWRGj91Qez_QMLvQNgWQ6lvqLNHmuAWaj7rFQCYwG0LkeNrP28qgUsmaGvxLkakJDvjAMBxnWXi-He5Nk-bMifUEikZl3hM5rDseBv5BbuZ0N_7DOAaUJ3LS1JNBIbN_11wr05kKRi0JG3j1yNsOd15Si-N24DYWrCgzrY4My2Aibv-jzbO_tHSg307p9PQAj9lSVwbMvb6PHFRz_MbHOCYabvuLvl47Le6lCy4JIWXHIb8jHReJmABCSAf231--00x6n8UayZfKBD_T-M6JX1gR3OhoNNtvu4q34q9pKTzPS-uKtE7uj-zDXb1JA8IfyMAkdXzkTjYNmlJrE4YmE-ho07_fiC8XlYE9mofx2gHC5Bjze-agH2jIaofbEtuVXGU6-M84UyIvCCJWzTO7-jKjmZHpFAbUm18Lj_Ow1sMJZlwvvP2HdrpkeLYx6d92g-o6xsuNhc0gxzatJDE60Gx8Or4QRfR5Ras-fsM6qIyF3fu1JRwJi2UNqwrNn9AJNSKa0n7N1dBV6bou91iTOkUbIacC4X8UBATF-mLlVv7NqVaw3rrUhtD0eS1xRjVn2P4Kg5yf6zEMquTJbYKL-uEZjNALg32u_Rr9APRayoBeIAKhlPlpwo0YEKariBYDooDNI_hh2-HlAn0ZVKJ8ugqBkaYIc4VMmqBmjQ.nQtGZi9sBcYxewmKlDTvtg",
  "requestedError": {
     "endpoint": "auth",
     "error": "invalid_request",
     "description": "a bad thing happened"
  }
}'

// Request to force an OAuth error from token endpoint
curl http://localhost:3003/api/authorize -v -d '{
  "clientId": "ipv-core-dev01",
  "request": "eyJjdHkiOiJKV1QiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiUlNBLU9BRVAtMjU2In0.G-7NVw-U-y_nuqhI_atQchaR62I2428w-jquIZnQkYkjmJJgS0V6-23TBwN3jNftrE3byisNFnje9zXf7hBBgr7UfnizoiSZyVufocDvt33CrHJdHuUEracOfXqhxwBFmmK0Qbwp6-FEKuxMQwUpDCH64r0M316jw7OycKIVmThA0amzhDhPWlVFqMT68GauwW6ajR8y9OYngzsUmZrBUBabZDdrZeqo8Hr95P0AwHoCD6i4EoTo6DPn9nl3L7Jyw1TaWguk0JnRPiU0MxLA8ks7ryXqXVBdx1breGxzUS4RMlbEX7mn5hPUDiD6wGlWbi2jPW8_5xMV-ZhoCQlufg.Ybu9nJra4aje5OEG.x4wrisYRGg43isZqbYRVP3EY8NpYmmffE_QJVbwuK3dsHPDqp5wsqcen3Dbg5GW--7OjzDpOnkoX3-oWJdnrL5n2Jk42OtXOGH_tUrR4qV2Vc9EdKE8WYvQieH1-VF5XoUQG-ltBk5B-tLZ18lC8w9Zy1PR_cfetYH35CcK7wAUaoX5cOw_aMoao2go2oW5DDD-9jj1yj0DjuOovswum-YFyZYmKizOxtLRiw1RsKR2KtQ5ZrWiac4KIAftLLJGn4gBVAkj6D9JbTacIl1hxQulqN4kSsZ0EKpTpnmNjJ1x8E6yqAT2uOEDjskg3gzD4wOwvrDh1sjjQywOOxbzE6W4uL8vcDn2pqhja-_M_FKsWWusSkCSMM5HYuo7-fOoRKRnLedKiD5uP3WWRGj91Qez_QMLvQNgWQ6lvqLNHmuAWaj7rFQCYwG0LkeNrP28qgUsmaGvxLkakJDvjAMBxnWXi-He5Nk-bMifUEikZl3hM5rDseBv5BbuZ0N_7DOAaUJ3LS1JNBIbN_11wr05kKRi0JG3j1yNsOd15Si-N24DYWrCgzrY4My2Aibv-jzbO_tHSg307p9PQAj9lSVwbMvb6PHFRz_MbHOCYabvuLvl47Le6lCy4JIWXHIb8jHReJmABCSAf231--00x6n8UayZfKBD_T-M6JX1gR3OhoNNtvu4q34q9pKTzPS-uKtE7uj-zDXb1JA8IfyMAkdXzkTjYNmlJrE4YmE-ho07_fiC8XlYE9mofx2gHC5Bjze-agH2jIaofbEtuVXGU6-M84UyIvCCJWzTO7-jKjmZHpFAbUm18Lj_Ow1sMJZlwvvP2HdrpkeLYx6d92g-o6xsuNhc0gxzatJDE60Gx8Or4QRfR5Ras-fsM6qIyF3fu1JRwJi2UNqwrNn9AJNSKa0n7N1dBV6bou91iTOkUbIacC4X8UBATF-mLlVv7NqVaw3rrUhtD0eS1xRjVn2P4Kg5yf6zEMquTJbYKL-uEZjNALg32u_Rr9APRayoBeIAKhlPlpwo0YEKariBYDooDNI_hh2-HlAn0ZVKJ8ugqBkaYIc4VMmqBmjQ.nQtGZi9sBcYxewmKlDTvtg",
  "credentialSubjectJson": "{\"passport\":[{\"expiryDate\":\"2030-01-01\",\"icaoIssuerCode\":\"GBR\",\"documentNumber\":\"321654987\"}],\"name\":[{\"nameParts\":[{\"type\":\"GivenName\",\"value\":\"Kenneth\"},{\"type\":\"FamilyName\",\"value\":\"Decerqueira\"}]}],\"birthDate\":[{\"value\":\"1965-07-08\"}]}",
  "evidenceJson": "{\"activityHistoryScore\":1,\"checkDetails\":[{\"checkMethod\":\"vri\"},{\"biometricVerificationProcessLevel\":3,\"checkMethod\":\"bvr\"}],\"validityScore\":2,\"strengthScore\":3,\"type\":\"IdentityCheck\"}",
  "resourceId": "something",
  "requestedError": {
     "endpoint": "token",
     "error": "invalid_request",
     "description": "a bad thing happened at the token endpoint"
  }
}'

// Request to force a 404 from credential endpoint
curl http://localhost:3003/api/authorize -v -d '{
  "clientId": "ipv-core-dev01",
  "request": "eyJjdHkiOiJKV1QiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiUlNBLU9BRVAtMjU2In0.G-7NVw-U-y_nuqhI_atQchaR62I2428w-jquIZnQkYkjmJJgS0V6-23TBwN3jNftrE3byisNFnje9zXf7hBBgr7UfnizoiSZyVufocDvt33CrHJdHuUEracOfXqhxwBFmmK0Qbwp6-FEKuxMQwUpDCH64r0M316jw7OycKIVmThA0amzhDhPWlVFqMT68GauwW6ajR8y9OYngzsUmZrBUBabZDdrZeqo8Hr95P0AwHoCD6i4EoTo6DPn9nl3L7Jyw1TaWguk0JnRPiU0MxLA8ks7ryXqXVBdx1breGxzUS4RMlbEX7mn5hPUDiD6wGlWbi2jPW8_5xMV-ZhoCQlufg.Ybu9nJra4aje5OEG.x4wrisYRGg43isZqbYRVP3EY8NpYmmffE_QJVbwuK3dsHPDqp5wsqcen3Dbg5GW--7OjzDpOnkoX3-oWJdnrL5n2Jk42OtXOGH_tUrR4qV2Vc9EdKE8WYvQieH1-VF5XoUQG-ltBk5B-tLZ18lC8w9Zy1PR_cfetYH35CcK7wAUaoX5cOw_aMoao2go2oW5DDD-9jj1yj0DjuOovswum-YFyZYmKizOxtLRiw1RsKR2KtQ5ZrWiac4KIAftLLJGn4gBVAkj6D9JbTacIl1hxQulqN4kSsZ0EKpTpnmNjJ1x8E6yqAT2uOEDjskg3gzD4wOwvrDh1sjjQywOOxbzE6W4uL8vcDn2pqhja-_M_FKsWWusSkCSMM5HYuo7-fOoRKRnLedKiD5uP3WWRGj91Qez_QMLvQNgWQ6lvqLNHmuAWaj7rFQCYwG0LkeNrP28qgUsmaGvxLkakJDvjAMBxnWXi-He5Nk-bMifUEikZl3hM5rDseBv5BbuZ0N_7DOAaUJ3LS1JNBIbN_11wr05kKRi0JG3j1yNsOd15Si-N24DYWrCgzrY4My2Aibv-jzbO_tHSg307p9PQAj9lSVwbMvb6PHFRz_MbHOCYabvuLvl47Le6lCy4JIWXHIb8jHReJmABCSAf231--00x6n8UayZfKBD_T-M6JX1gR3OhoNNtvu4q34q9pKTzPS-uKtE7uj-zDXb1JA8IfyMAkdXzkTjYNmlJrE4YmE-ho07_fiC8XlYE9mofx2gHC5Bjze-agH2jIaofbEtuVXGU6-M84UyIvCCJWzTO7-jKjmZHpFAbUm18Lj_Ow1sMJZlwvvP2HdrpkeLYx6d92g-o6xsuNhc0gxzatJDE60Gx8Or4QRfR5Ras-fsM6qIyF3fu1JRwJi2UNqwrNn9AJNSKa0n7N1dBV6bou91iTOkUbIacC4X8UBATF-mLlVv7NqVaw3rrUhtD0eS1xRjVn2P4Kg5yf6zEMquTJbYKL-uEZjNALg32u_Rr9APRayoBeIAKhlPlpwo0YEKariBYDooDNI_hh2-HlAn0ZVKJ8ugqBkaYIc4VMmqBmjQ.nQtGZi9sBcYxewmKlDTvtg",
  "credentialSubjectJson": "{\"passport\":[{\"expiryDate\":\"2030-01-01\",\"icaoIssuerCode\":\"GBR\",\"documentNumber\":\"321654987\"}],\"name\":[{\"nameParts\":[{\"type\":\"GivenName\",\"value\":\"Kenneth\"},{\"type\":\"FamilyName\",\"value\":\"Decerqueira\"}]}],\"birthDate\":[{\"value\":\"1965-07-08\"}]}",
  "evidenceJson": "{\"activityHistoryScore\":1,\"checkDetails\":[{\"checkMethod\":\"vri\"},{\"biometricVerificationProcessLevel\":3,\"checkMethod\":\"bvr\"}],\"validityScore\":2,\"strengthScore\":3,\"type\":\"IdentityCheck\"}",
  "resourceId": "something",
  "requestedError": {
    "userInfoError": "404"
  }
}'
```

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYI-5874](https://govukverify.atlassian.net/browse/PYI-5874)

Wynndow committed:
PYIC-6182: Use domain object for auth response

This updates the AuthHandler#generateResponse method to parse a domain
object from the request to use, rather than accessing the query params
directly. This is to allow us to use a similar domain object when
handling an API request to the headless endpoint. The params receieved
in the usual flow and the API endpoint will be slightly different, and
using an object should mean we can reuse the same logic for generating
response on both modes.

This also removes all the logic around expiry dates for VCs, as we don't
use them.

Wynndow committed:
PYIC-5874: Add new api endpoint

This adds a new endpoint for API based auth requests. An AuthRequest is
deserialized from the incoming JSON body, and then fed into the same
logic that the form based requests use.

The logic has been extracted to a method so it can be shared.

There's been some renaming of the fields in the AuthRequest object, and
some null checks added, but it's mostly exactly the same logic.
	modified:   di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/handlers/AuthorizeHandler.java

Joe-Edwards-GDS commented on di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/domain/AuthRequest.java:
```diff
@@ -0,0 +1,142 @@
+    private String fraudScore;
+    private String verificationScore;
+    private String biometricVerificationScore;
+    private String evidenceJson;
```
We can probably streamline this a bit for the API version - e.g. we don't really need all these individual scores, CIs, etc., we just need the overall evidence JSON block.

It might also be nice to structure some of the other properties (e.g. all the F2F stuff, the error stuff) to make it easier to work with?

No need to overthink it, but it'll be much easier to change now than later!

Joe-Edwards-GDS commented on di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/handlers/AuthorizeHandler.java:
```diff
@@ -260,278 +257,164 @@ public AuthorizeHandler(
+    public final Route apiAuthorize =
+            (Request request, Response response) ->
+                    generateResponse(
+                            OBJECT_MAPPER.readValue(request.body(), AuthRequest.class), response);
```
Do we need to validate the JAR here?

Wynndow replied:
I don't think so. We'd just be validating that the thing we're passing in ourselves is correct. And we go on to decrypy the jar, and then extract the signed jwt. And then pull out all the details from there. If anything is wrong it'll blow up at some point and we'd see some failing tests.

Joe-Edwards-GDS reviewed (commented):
I haven't looked at this in any detail, but looks like a broadly sensible approach!

Wynndow committed:
PYIC-5874: Set content type and add tests

This updates the api auth endpoint to set the content type of the
response to application/json

It also renames cis to ci to be more inline with the evidence json.

Also now loads the F2F queue URL dynamically, rather than from a static
constant. It's easier to test.

Also now uses an http client to send message to the F2F queue rather
than the nimbus lib. Easier to test and more consistant.

Adds a bunch of tests for the api auth endpoint, and refactors the
structure of the tests to hopefully make them more obvious.

Wynndow committed:
PYIC-5874: Only generate nbf if required

This updates the logic to only try an construct a NMF from the form data
if the flag has been set.

Wynndow committed:
PYIC-5874: Split AuthRequest into records

This converts the AuthRequest class into two separate records, one for
form auth requets and one for api auth requests. This keeps the api auth
request record cleaner.

It also refactors the handler to create a new method for generating the
evidence map from the records, as the two types now need handling
slightly differently.

Wynndow committed:
PYIC-5874: Make mitigations a structured field

This moves the details for mitigations into a structured object. This
will mean that API requests will need to nest some JSON to define
mitigations.

Wynndow committed:
PYIC-5874: Make F2F a structured field

This moves the details for f2f requests into a structured object. This
will mean that API requests will need to nest some JSON to define
f2f details.

Wynndow committed:
PYIC-5874: Make requested error a structured field

This moves the details for a requested error into a structured object.
This will mean that API requests will need to nest some JSON to define
an error.

Wynndow committed:
PYIC-5874: Make gpg45 scores a structured field

This moves the gpg45 scores for a form request to a structured object.
It keeps the FormAuthRequest record a bit cleaner, and hopefully easier
to work with.

MikeCollingwood commented on di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/config/CredentialIssuerConfig.java:
```diff
@@ -9,39 +9,15 @@ public class CredentialIssuerConfig {
-    public static final String F2F_STUB_QUEUE_URL = getConfigValue("F2F_STUB_QUEUE_URL", null);
-    public static final String F2F_STUB_QUEUE_NAME = getConfigValue("F2F_STUB_QUEUE_NAME", null);
+    public static final String F2F_STUB_QUEUE_NAME_DEFAULT =
+            getConfigValue("F2F_STUB_QUEUE_NAME", null);
```
Can this exclude the second parameter with the `getConfigValue` update below?

MikeCollingwood commented on di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/domain/Credential.java:
```diff
@@ -7,21 +7,18 @@ public class Credential {
     private final Map<String, Object> evidence;
     private final String userId;
     private final String clientId;
-    private final Long exp;
```
Doesn't the stub have an input for this expiry?

Wynndow replied:
It does, but it can be removed as CRIs no longer set `exp`, or at least shouldn't. And we don't use it for anything - we use `nbf` for doing expiry stuff. So this PR pulls out the expiry stuff to keep it simpler overall.

The next step would be to remove all the stuff in the e2e tests that interacts with the expiry stuff in the CRI stub. And then finally remove those options from the CRI stub frontend.

MikeCollingwood replied:
Ok, good to know - I guess the cleanup is in a separate ticket?

Wynndow replied:
It will be...
https://govukverify.atlassian.net/browse/PYIC-6375

MikeCollingwood commented on di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/handlers/AuthorizeHandler.java:
```diff
@@ -263,277 +267,148 @@ public AuthorizeHandler(
+        String redirectUri =
+                signedJWT.getJWTClaimsSet().getClaim(RequestParamConstants.REDIRECT_URI).toString();
+        String userId = signedJWT.getJWTClaimsSet().getSubject();
+        String state = signedJWT.getJWTClaimsSet().getClaim(RequestParamConstants.STATE).toString();
```
Does this want linting?

Wynndow replied:
Believe it or not this _is_ linted. Thank you Spotless...

MikeCollingwood reviewed (commented):
I see the use of the interface and modification of generate response to take it as the parameter. In addition, there were some useful refactors, e.g. `generateNbf` & `generateEvidenceMap`, cutting down the `generateResponse` method

There are a number of refactors made, so would feel more comfortable if it were tested

MikeCollingwood reviewed (approved):
LGTM!
